plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -271,9 +271,23 @@ private CompletableFuture<ConnectorSplitBatch> getNextBatchInternal(int maxSize)
                 }
 
                 if (recordScannedFiles) {
```

Comment:
potential follow-up: `io.trino.plugin.iceberg.IcebergTableHandle#forOptimize` takes boolean `recordScannedFiles` and it is used only once with the value `true` - parameter could be removed from the method.